### PR TITLE
Vision circular error

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ module.exports = (Hapi, options, allDone) => {
     },
     server: ['config', (done, result) => {
       const config = result.config;
-      const serverConfig = _.cloneDeep(config.server || {});
```
Comment (jgallen23) [inline index.js]: why was this removed?

Comment (jgallen23) [inline index.js]: I think we want to keep just cloning the server config because the engine gets set.  Basically whenever we have a `require()` we need to do a clone of that so that the config object doesn't get that stuff

Comment (jgallen23) [inline index.js]: I don't see the cloneDeep on the view either...

**Comment by orthagonal** [comment]
Cloning config seems to be the key to avoiding the circular error.

File: index.js [hunk 2]
```diff
@@ -76,7 +77,7 @@ module.exports = (Hapi, options, allDone) => {
       if (reporters.length !== 0) {
         config.logging.reporters = reporters;
         server.register({
-          register: requireCwd('good'),
+          register: _.cloneDeep(requireCwd('good')),
```
Comment (jgallen23) [inline index.js]: huh?

**Comment by jgallen23** [comment]
I think there was a misunderstanding.  why are you cloning the require statements?

**Comment by jgallen23** [comment]
the config object needs to be cloned, not the require statement.  

for example:

```const views = _.clone(config.views);```
then pass that to `server.views(views)`

**Comment by jgallen23** [comment]
I think I'm just going to do this myself.  clearly not communicating what I want very well
